plasmapy/atomic/ionization_state.py

```diff
@@ -327,18 +378,19 @@ def n_elem(self, value: u.m ** -3):
     @u.quantity_input
     def number_densities(self) -> u.m ** -3:
         """Return the number densities for each state."""
-        if not np.isnan(self._n_elem):
+        try:
             return (self.n_elem * self.ionic_fractions).to(u.m ** -3)
-        else:
-            raise AtomicError(
-                "Insufficient information to return number densities.")
+        except Exception:
+            return np.full(self.atomic_number + 1, np.nan) * u.m ** -3
```

Comment:
Is there any way we could potentially make this more selective than the broad `except Exception` clause?